examples/helloworld/go.mod

```diff
@@ -0,0 +1,5 @@
+module google.golang.org/grpc/examples/helloworld
```

Comment:
Will the quick start still replace this `go.mod` file with another `go.mod` after copying it out?

Do the imports need to be changed in this PR to be relative to this `go.mod`'s directory?

Reply:
> Will the quick start still replace this go.mod file with another go.mod after copying it out?

No, that wasn't required when I tested this locally.

> Do the imports need to be changed in this PR to be relative to this go.mod's directory?

Based on what I've read, this doesn't seem to be the case, but I guess that we can't really test it out until a new release / tag is published.

I could change the constraint to require, say, `v1.29.0-dev.1` that way we could consider testing it on a dev release. WDYT?

Reply:
Oh, right..  I see the difference between what this does and what we have today.  The `go.mod` file being copied is the key point.

If we were to copy the whole `grpc` tree from the `pkg` directory in your example, or just `git clone` the repo, then we wouldn't have the problems you ran into about conflicting import paths.

Reply:
That would seem right, but I did a quick test now (with a `git clone` of the `grpc-go` repo), and I'm running into issues. But maybe that's just because I have too many partial setups. (I can try again later if this is an option that you'd like to consider).

Reply:
I wasn't trying to suggest this was preferred, but that it _should_(?) work due to your local go.mod fully subsuming the remote go.mod, and go recognizing that and using the local files instead of the downloaded package's files.